uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/controller/UniProtKBController.java

```diff
@@ -218,6 +227,30 @@ public ResponseEntity<MessageConverterContext<UniProtKBEntry>> getByAccession(
         }
     }
 
+    private ResponseEntity<MessageConverterContext<UniProtKBEntry>> redirectToEntryVersion(
+            String accessionOrId, HttpServletRequest request) {
+        String version = null;
+        if (request.getParameter("version").equals("last")) {
+            try {
+                String entryVersionHistoryURI = "https://rest.uniprot.org/unisave/" + accessionOrId;
+                String response = Utils.httpGetRequest(entryVersionHistoryURI);
```

Comment:
I think we need to be prepared for an accession that does not exist. Here we are assuming that all accessions received as a parameter exists in UniSave

Reply:
This error response will break your code in line 240. because your code is always expecting "results" in the json, and when you receive back a json with error message, it will break